src/core/manager.c

```diff
@@ -2749,6 +2749,10 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
                 r = unit_deserialize(u, f, fds);
                 if (r < 0)
                         goto finish;
+
+                r = unit_realize_cgroup(u);
```

Comment:
Hm... a tough case. I like the simplicity of your patch, but it seems more correct to not update the cgroup configuration on reload. In particular the cgroup configuration could be tied to the details of the ExecStart command (e.g. imagine a unit which had MemoryMax=FOO, ExecStart=/bin/java -XmFOO, and then is changed to MemoryMax=BAR, ExecStart=/bin/java -XmBAR. It is expected that on unit restart you get the new value, but getting the new limits on an unrelated daemon-reload could be surprising). Hence I think we should find a way to fix the original issue without changing this.